### PR TITLE
Expose a few functions from fpdf_text.h for getting text on a page.

### PR DESCRIPTION
**Pull request by billmoyers**
Hi, this exposes some of the FPDFText_* functions which are useful for getting text. Thanks.

### CODE REVIEW

**Comment by barteksc** [comment]
What is the purpose of `nativeBitmapFillRect()` function? You can fill bitmap with color using Android API

**Comment by billmoyers** [comment]
My impression was that FPDFBitmap_FillRect would fill in between the text and the background layer which I believe would be difficult to do after the bitmap has been rendered by FPDF.  Maybe I missed something?

**Comment by barteksc** [comment]
There are no layers, this just a bitmap. FillRect will fill given region with given color

**Comment by billmoyers** [comment]
I just realized that I didn't mean to include the FillRect in the pull request anyway because I hadn't really tested it.  I removed the FillRect commit from this PR.  The other commit was tested.  How does that one look?

**Comment by barteksc** [comment]
Thank you for your work, I will review commit source and add it to next release

File: src/main/java/com/shockwave/pdfium/PdfiumCore.java [hunk 1]
```diff
@@ -440,4 +466,201 @@ public RectF mapRectToDevice(PdfDocument doc, int pageIndex, int startX, int sta
...
+                e.printStackTrace();
+            } catch (Exception e) {
+                Log.e(TAG, "Exception throw from native");
+                e.printStackTrace();
+            }
+        }
+        return 0;
+    }
+
+    public class Rect {
```
Comment (johngray1965) [inline src/main/java/com/shockwave/pdfium/PdfiumCore.java]: Other places seem to using android's RectF?

**Review by johngray1965 — COMMENTED** [review]
Could we also expose the device to page mapping functions?

**Comment by billmoyers** [comment]
@johngray1965 is right, we don't need the extra class. I removed it for RectF and pushed the commit.

**Comment by envolva** [comment]
Is this feature still in the Works? We would like to use this with Pdfium. Thanks.

**Comment by johngray1965** [comment]
I've been using this branch, I'd love to see it get merged.

**Comment by oceanwaves90** [comment]
1. Is it needed to clear mNativeTextPagesPtr and call nativeCloseTextPages for each of them when disposing?

2. And I wonder why openTextPage calls openPage also. Are textPages different from usual pages?(why does it load usual pages too?)

3. What does 'length' mean in textPageGetBoundedText? How long do you recommend for me?

**Pull request closed without merging**